extensions/objects/templates/svg-filename-editor.component.ts

```diff
@@ -637,14 +730,21 @@ angular.module('oppia').component('svgFilenameEditor', {
               action: 'remove',
               object: undoObj.object
             });
-            // Not adding the shape to canvasObjects because it is added by the
-            // event function.
+            // Adding the object in the correct position according to initial
+            // order.
+            ctrl.undoFlag = true;
+            ctrl.canvasObjects.splice(undoObj.index, 0, undoObj.object);
             ctrl.canvas.add(undoObj.object);
           }
           ctrl.canvas.renderAll();
         }
       };
 
+      ctrl.isUndoEnabled = function() {
+        return Boolean(
```

Comment:
Do we need to type cast this? ditto for similar uses below.

Reply:
I have removed it. Done.